[CFDP-1087] Group Card

dzwood opened the pull request:
## GroupCard
* Added `GroupCard` to `ui-kit`
* Added `SquareAvatar`
* Updated `useGroup` hook
* Added `textTrimmer` to `utils`

drewbarontini reviewed (changes requested):
Nice work, @dzwood! 🎉 

I still haven't run and tested this locally, but want to get this review in for some comments on the code changes 👍

dzwood commented on ui-kit/GroupCard/GroupCard.js:
```diff
@@ -0,0 +1,132 @@
+  const avatarsDiff = props.totalAvatars - props.avatars.length;
+
+  return (
+    <Styled display="flex" flexDirection="column" {...props}>
```
Gotcha yeah that makes sense! In `GroupCard.styles` I added a `GroupCard` style that overrides the default `Card` with `styled(Card)` to fix that. Then in the `GroupCard` component instead of `Styled` I'm now using `Styled.GroupCard`. Let me know if that works! Sorry still getting the hang of `styled-components` 😅

dzwood commented:
@drewbarontini Thank you for the review and comments! I put in the requested changes, let me know if there is anything else 😊

drewbarontini commented on ui-kit/GroupCard/GroupCard.styles.js:
```diff
@@ -57,6 +60,7 @@ const GroupCardTitle = styled.div`
 Card.GradientBackground = GradientBackground
+Card.GroupCard = GroupCard
 Card.GroupCardContent = GroupCardContent
 Card.GroupCardTitle = GroupCardTitle
```
I didn't notice this before, but these should be `GroupCard` so then `Styled` is set to `GroupCard` at the top, which means you don't have to do `Styled.GroupCard` then.

drewbarontini commented on ui-kit/GroupCard/GroupCard.js:
```diff
@@ -7,15 +7,15 @@ import Styled from './GroupCard.styles';
 
   return (
-    <Styled display="flex" flexDirection="column" {...props}>
+    <Styled.GroupCard {...props}>
```
This will just be `<Styled>` once we make the changes discussed in `GroupCard.styles.js`.

drewbarontini commented on ui-kit/GroupCard/GroupCard.styles.js:
```diff
@@ -42,6 +40,11 @@ const GradientBackground = styled.div`
   position: relative;
 `;
 
+const GroupCard = styled(Card)`
```
Instead of doing `styled(Card)`, you can also just do `as={Card}` on `<Styled>`. But this also works fine, too 👍

dzwood replied:
So I would add `as={Card}` to `<Styled>` inside of the `GroupCard.js`?

drewbarontini replied:
Yes, you can do that, but it's probably better to just leave this how it is 👌

drewbarontini replied:
The only other thing I'd say we change is to move `GroupCard` to the top of this file since that's the default component being exported.

drewbarontini reviewed (changes requested):
Looking good! A few more notes 👍

drewbarontini commented:
@dzwood It looks like there are some unused imports. If you click on `Update branch` above, you'll get the latest commit I added on `main` that fixes an issue with `yarn lint`. You should then be able to run it — or look in your editor — to see the imports that are unused (and other ESLint warnings) 👍

dzwood commented:
@drewbarontini I updated from **main** running ES Lint and cleaned up all the unused imports for the `GroupCard`. The rest of the the linter warnings seem to be coming from the `.stories.js` files. I'm assuming it's because they don't use an `export default`, is that something we want to have the linter ignore?

drewbarontini reviewed (approved):
Very nice work, @dzwood! 🎉
